File: adapters/database/src/main/java/net/onelitefeather/pandorascluster/database/models/player/LandPlayerEntity.java
package net.onelitefeather.pandorascluster.database.models.player;

import jakarta.persistence.*;
import net.onelitefeather.pandorascluster.dto.player.LandPlayerDto;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.Objects;

@Entity
@Table(name = "land_players")
public final class LandPlayerEntity implements LandPlayerDto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "uuid", length = 36, nullable = false, unique = true)
    private String uuid;

    @Column(name = "name", length = 16, nullable = false)
    private String name;

    public LandPlayerEntity() {
        // Empty constructor for Hibernate
    }

    public LandPlayerEntity(Long id, String uuid, String name) {
        this.id = id;
        this.uuid = uuid;
        this.name = name;
    }

    @Override
    public @Nullable Long id() {
        return id;
    }

    @Override
    public @NotNull String uuid() {
        return uuid;
    }

    @Override
    public @NotNull String name() {
        return name;
    }

    @Override
    public final boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof LandPlayerEntity that)) return false;

        return Objects.equals(id, that.id) && Objects.equals(uuid, that.uuid) && Objects.equals(name, that.name);
    }

    @Override
    public int hashCode() {
        int result = Objects.hashCode(id);
        result = 31 * result + Objects.hashCode(uuid);
        result = 31 * result + Objects.hashCode(name);
        return result;
    }
}

File: adapters/database/src/main/kotlin/net/onelitefeather/pandorascluster/database/models/player/LandPlayerEntity.kt
package net.onelitefeather.pandorascluster.database.models.player

import jakarta.persistence.*
import net.onelitefeather.pandorascluster.dbo.player.LandPlayerDBO

@Entity
@Table(name = "land_players")
data class LandPlayerEntity(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(columnDefinition = "VARCHAR(36)")
    val uuid: String? = null,

    @Column
    val name: String? = null
): LandPlayerDBO {

    constructor() : this(null, "", "")

    @Override
    override fun toString(): String {
        return this::class.simpleName + "(id = $id , uuid = $uuid , name = $name )"
    }

    override fun id(): Long? = id

    override fun uuid(): String = uuid ?: ""

    override fun name(): String = name ?: ""

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is LandPlayerEntity) return false

        if (id != other.id) return false
        if (uuid != other.uuid) return false
        if (name != other.name) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id.hashCode() ?: 0
        result = 31 * result + uuid.hashCode()
        result = 31 * result + (name?.hashCode() ?: 0)
        return result
    }
}